IdentityCore/src/cache/crypto/MSIDAssymetricKeyPair.m

```diff
@@ -218,6 +231,36 @@ - (nullable NSData *)decrypt:(nonnull NSString *)encryptedMessageString
     }
 }
 
+- (NSString *)signData:(NSString *)message
+{
+    NSData *signedData = [[message dataUsingEncoding:NSUTF8StringEncoding] msidSignHashWithPrivateKey:self.privateKeyRef];
+    return [[NSString alloc] initWithData:signedData encoding:NSUTF8StringEncoding];
+}
+
+- (NSDate *)creationDate
+{
+    if (!_creationDate)
+    {
+        NSDictionary* privateKeyQuery = @{ (id)kSecValueRef: (__bridge id)self.privateKeyRef,
```

Comment:
This will crash if privateKeyRef is nil. Is it always non nil here?

Reply:
As discussed. privateKeyRef is set in the constructor and will always be non-nil